MoreLinq/Locf.cs

```diff
@@ -0,0 +1,113 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Atif Aziz. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Returns the source sequence with null values replaced with last
+        /// the value seen in the sequence.
+        /// </summary>
+        /// <param name="source">The source sequence.</param>
+        /// <typeparam name="T">Type of the elements in the source sequence.</typeparam>
+        /// <returns>
+        /// An <see cref="IEnumerable{T}"/> with missing values replaced.
+        /// </returns>
+        /// <remarks>
+        /// This method uses deferred execution semantics and streams its
+        /// results. If values are null at the start of the sequence then
+        /// they remain missing.
+        /// </remarks>
+
+        public static IEnumerable<T?> Locf<T>(this IEnumerable<T?> source)
```

Comment:
Hmmm. Thinking out loud here. Maybe there could be a version that returns an `IEnumerable<T>` (ie, not nullable)? In that case, what should happen with leading `null`s? One could add a `seed` parameter, or error out. I would think the latter would be better as it is trival to prepend a seed using `Prepend`.